analyzers/src/SonarAnalyzer.CSharp/Rules/NotAssignedPrivateMember.cs

```diff
@@ -87,56 +82,65 @@ protected override void Initialize(SonarAnalysisContext context)
                     var usedMemberSymbols = new HashSet<ISymbol>(usedMembers.Select(tuple => tuple.Symbol));
 
                     var assignedMemberSymbols = GetAssignedMemberSymbols(usedMembers);
+                    var unassignedUsedMemberSymbols = allCandidateMembers.Where(x => usedMemberSymbols.Contains(x.Symbol) && !assignedMemberSymbols.Contains(x.Symbol));
 
-                    foreach (var candidateMember in allCandidateMembers)
+                    foreach (var candidateMember in unassignedUsedMemberSymbols)
                     {
-                        if (!usedMemberSymbols.Contains(candidateMember.Symbol))
-                        {
-                            /// reported by <see cref="UnusedPrivateMember"/>
-                            continue;
-                        }
-
-                        if (!assignedMemberSymbols.Contains(candidateMember.Symbol))
-                        {
-                            var field = candidateMember.SyntaxNode as VariableDeclaratorSyntax;
-                            var property = candidateMember.SyntaxNode as PropertyDeclarationSyntax;
+                        var field = candidateMember.SyntaxNode as VariableDeclaratorSyntax;
+                        var property = candidateMember.SyntaxNode as PropertyDeclarationSyntax;
 
-                            var memberType = field != null ? "field" : "auto-property";
+                        var memberType = field != null ? "field" : "auto-property";
 
-                            var location = field != null
-                                ? field.Identifier.GetLocation()
-                                : property.Identifier.GetLocation();
+                        var location = field != null
+                            ? field.Identifier.GetLocation()
+                            : property.Identifier.GetLocation();
 
-                            c.ReportDiagnosticIfNonGenerated(Diagnostic.Create(rule, location, memberType, candidateMember.Symbol.Name));
-                        }
+                        c.ReportDiagnosticIfNonGenerated(Diagnostic.Create(Rule, location, memberType, candidateMember.Symbol.Name));
                     }
                 },
                 SymbolKind.NamedType);
+
+        private static List<SyntaxNodeSymbolSemanticModelTuple<SyntaxNode, ISymbol>> RetrieveFielAndPropertyDeclarations(CSharpRemovableDeclarationCollector removableDeclarationCollector)
+        {
+            var candidateFields = removableDeclarationCollector.GetRemovableFieldLikeDeclarations(
```

Comment:
these are a bit hard to read. I suggest to move the parameters on the same line with the method call and align the `.Where` under the `.GetRemovableFieldLikeDeclarations`. Also && should be on the next line.

Same for `candidateProperties` below.

Reply:
Actually, I want to bring up this alignment topic on our next how we work meeting. I don't feel it comfortable reading those aligned lines, where this dot part goes almost to the right part of the screen.

Reply:
This also made me turning off type hints functionality of Rider completely, to make it at least readable

Reply:
(basically, with Inlay hints on this alignment makes no sense, since it's not aligned in any case.

Reply:
I find the inlay hints rarely useful. For me they make the code harder to read since they change indentation.

Reply:
For me the thing is not only in them, rather reading the code in the right part of the screen. So I bet this is a good topic to discuss within our team.

Reply:
We had a discussion about this in the past but sure, we can bring it up and see how we can improve this.